packages/experimental/src/rest/BaseResource.ts

```diff
@@ -236,21 +234,24 @@ export default abstract class BaseResource extends EntityRecord {
       }),
     );
   }
-}
 
-// We're only allowing this to get set for descendants but
-// by default we want Typescript to treat it as readonly.
-Object.defineProperty(BaseResource.prototype, 'url', {
-  get(): string {
-    if (this.__url !== undefined) return this.__url;
-    // typescript thinks constructor is just a function
-    const Static = this.constructor as typeof BaseResource;
-    return Static.url(this);
-  },
-  set(url: string) {
-    this.__url = url;
-  },
-});
+  static {
+    Object.defineProperty(this.prototype, 'url', {
```

Comment:
I’m confused why this isn’t defined as a normal prototype getter/setter?

Reply:
Typescript distinguishes between getters but I don't want the types to be that way

Reply:
Otherwise a user cannot declare a non getter type, which is exactly what is enabled by the setter definition.

Reply:
It seems very unfortunate to write worse actual code just to make TS types happy.

Reply:
Are there any downsides other than just looking uglier?